grouper/fe/handlers/group_join.py

```diff
@@ -173,15 +186,13 @@ def _get_member(self, member_choice):
 
         return self.session.query(resource).filter_by(name=member_name, enabled=True).one()
 
-    def _get_choices(self, group):
-        # type: (Group) -> List[Tuple[str, ...]]
-        # This returns List[Tuple[str, str]], but mypy is confused by the * 2 syntax.
+    def _get_choices(self, group, members, user_is_member):
```

Comment:
I think the logic would be a bit clearer here if instead of `members` we passed in only the group members, as right now it's not super obvious until you read through the code why `user_is_member` is separate. Would also make the type and usage of `members` much simpler (could be as simple as `Set[str]` with just a single line of preprocessing)